clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/SearchController.groovy

```diff
@@ -45,15 +48,28 @@ class SearchController {
    */
   @PreAuthorize("@fiatPermissionEvaluator.storeWholePermission()")
   @RequestMapping(value = '/search', method = RequestMethod.GET)
-  List<SearchResultSet> search(SearchQueryCommand q) {
+  List<SearchResultSet> search(@RequestParam(value = "q", defaultValue = "", required = false) String query,
+                               @RequestParam(value = "type") List<String> type,
+                               @RequestParam(value = "platform", required = false) String platform,
+                               @RequestParam(value = "pageSize", defaultValue = "10000", required = false) int pageSize,
+                               @RequestParam(value = "page", defaultValue = "1", required = false) int page,
+                               HttpServletRequest httpServletRequest) {
+
+    def filters = httpServletRequest.getParameterNames().findAll { String parameterName ->
```

Comment:
I don't love this at all but was torn with whether or not to have the ability to test the clouddriver endpoint directly - which still seems like a useful feature.  Unfortunately, there doesn't appear to be a way in spring-web to have explicit `@RequestParams` set and have any remaining parameters placed into a map.  This filter/collect pattern is already used elsewhere in gate.